### DISCUSSION TITLE
Multiple Proxy functionality

### DISCUSSION THREAD

**Discussion by sporetec** [opsdisk/pagodo#69] [General]
Hi everyone, not really a issue so im starting a conversation.
First of all great work on this and the underlying yagooglesearch!

Im looking into auto spawning some azure containers to use as the proxies for pagodo but noticed the following:
When running pagodo with a list of proxies and its running into a 429 situation the script is sleeping 60 mins as if it had only one IP, is this intended? Am i missing something?
I tried reducing the max and min wait times to the time divided by my number of proxies available but then some random dorks end up in a 429. Which I guess is caused by also having a smaller wait between paging?

How can i circumvent this or configure this correctly? As right now i don't see the value of actually using multiple proxies vs one proxy.

Hope everything is clear, if not I'm more than happy to explain with some code examples!
Thanks in advance!

**Comment by opsdisk** [comment]
Hi @sporetec - To start off, are you using the Azure containers as proxies right now?  If so, could you provide some sort of validation that your Azure containers are round robin-ing through the proxies correctly?  You could do that with a tool like proxychains and editing the round robin section in the config at this line: https://github.com/rofl0r/proxychains-ng/blob/master/src/proxychains.conf#L27  So something like:

```sh
proxychains curl https://icanhazip.com
```

with the config file having your list of proxy servers.  Let's validate that the exit/public IP of your Azure containers is actually changing.  If they are all egressing out the same public IP, you'll get the HTTP 429 pretty quickly.

**Comment by sporetec** [comment]
Ok this is actually akward.
I tried it again for testing and proxychains is producing some weird output.
However when then trying to check the output with pagodo and wireshark i found that: This is working as intended BUT azure is actually reusing some old IPs which is something i got to solve myself.

Thanks for your help, really appreciated!

**Reply by opsdisk** [reply]
No worries...I'll keep the discussion open for a few weeks in case it is something with pagodo/yagooglesearch.
